modules/navier_stokes/src/postprocessors/MassFluxWeightedFlowRate.C

```diff
@@ -58,11 +58,19 @@ MassFluxWeightedFlowRate::computeFaceInfoIntegral([[maybe_unused]] const FaceInf
   const bool correct_skewness =
       _advected_interp_method == Moose::FV::InterpMethod::SkewCorrectedAverage;
 
-  const auto face_arg = Moose::FaceArg({fi,
-                                        Moose::FV::limiterType(_advected_interp_method),
-                                        MetaPhysicL::raw_value(vel) * fi->normal() > 0,
-                                        correct_skewness,
-                                        nullptr});
+  mooseAssert(_adv_quant->hasFaceSide(*fi, true) || _adv_quant->hasFaceSide(*fi, true),
+              "Advected quantity must be defined on one of the sides of the face!");
+  mooseAssert((_adv_quant->hasFaceSide(*fi, true) == _density.hasFaceSide(*fi, true)) ||
+                  (_adv_quant->hasFaceSide(*fi, false) == _density.hasFaceSide(*fi, false)),
+              "Density must be define at least on one of the sides where the advected quantity is "
```

Comment:
I can add that assert too, I can't think of a case where that would not be the case.

Reply:
Isn't this going to test for that too? If one is defined on one side only and the other on the other this will be triggered. That means that the density will be defined on the block which is used for the advecting functor.